### PR TITLE
Prepare for prerelease flow and introduce documentation and tooling for emergency releases

### PR DESCRIPTION
**Pull request by Rugvip**
Co-authored-by: Johan Haals <johan.haals@gmail.com>

## Hey, I just made a Pull Request!

We're preparing to move over our release process to include prereleases. As part of that we want to maintain the possibility of shipping emergency releases when required, either for security fixes or broken packages. The changeset flow requires a bit of extra tooling to bump up the "current" version during a prerelease, which is what the script is about.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/publishing.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+**This emergency release process is intended only for the Backstage
+maintainers.**
+
+For this example we will be using the `@backstage/plugin-foo` package as an
+example and assume that it is currently version `1.5.0` in the master branch.
+
+In the event of a severe bug being introduced in version `1.5.0` of the
+`@backstage/plugin-foo` released in the `2048-01-01` release, the following
+processed is used to release an emergency fix as `1.5.1`:
```
Comment (freben) [inline docs/publishing.md:26]:
```suggestion
process is used to release an emergency fix as `1.5.1`:
```

File: docs/publishing.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+- [ ] Identify the release that needs to be patched, in this case we're fixing a
+      broken release, so it would be the most recent one, `2048-01-01`. In the
+      event of a backported security fix, the release that has the last
+      published version of each major version of the package should be the one
+      patched.
+- [ ] Make sure a patch branch exists for the release that is being patched. If
+      a patch already exists, reuse the existing branch.
```
Comment (freben) [inline docs/publishing.md:34]:
```suggestion
      a patch already exists, reuse the existing branch. The branch _must_ always be named exactly `release-<release>-patch`.
```

File: docs/publishing.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  ```
+
+- [ ] With the `release-2048-01-01-patch` branch as a base, create a new branch
+      for your fix:
+
+  ```bash
+  git checkout -b ${USER}/release-2048-01-01-emergency-fix
+  ```
+
+- [ ] Apply fixes and create a new patch changeset for the effected package,
```
Comment (freben) [inline docs/publishing.md:49]:
```suggestion
- [ ] Apply fixes and create a new patch changeset for the affected package,
```

File: docs/publishing.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      then commit these changes.
+- [ ] Run `yarn release` in the root of the repo in order to convert your
+      changeset into package version bumps and changelog entries. Commit these
+      changes as a second `"Generated release"` commit.
+- [ ] Create PR towards the base branch(`release-2048-01-01-patch`) containing
```
Comment (freben) [inline docs/publishing.md:54]:
```suggestion
- [ ] Create PR towards the base branch (`release-2048-01-01-patch`) containing
```

File: docs/publishing.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      a patch already exists, reuse the existing branch.
+
+  ```bash
+  git checkout release-2048-01-01
+  git checkout -b release-2048-01-01-patch
+  git push --set-upstream origin release-2048-01-01-patch
+  ```
+
+- [ ] With the `release-2048-01-01-patch` branch as a base, create a new branch
+      for your fix:
```
Comment (freben) [inline docs/publishing.md:43]:
```suggestion
      for your fix. This branch can be named anything, but the following naming pattern may be suitable:
```

File: docs/publishing.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  ```
+
+- [ ] Apply fixes and create a new patch changeset for the effected package,
+      then commit these changes.
+- [ ] Run `yarn release` in the root of the repo in order to convert your
+      changeset into package version bumps and changelog entries. Commit these
+      changes as a second `"Generated release"` commit.
+- [ ] Create PR towards the base branch(`release-2048-01-01-patch`) containing
+      the two commits.
+- [ ] Review/Merge PR into `release-2048-01-01-patch`. This will automatically
```
Comment (freben) [inline docs/publishing.md:56]:
```suggestion
- [ ] Review/Merge the PR into `release-2048-01-01-patch`. This will automatically
```

File: docs/publishing.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      trigger a release.
+- [ ] In the master branch, add a `.changeset/patched.json` to make sure that
+      future releases of the packages are bumped accordingly:
```
Comment (freben) [inline docs/publishing.md:59]: How does one go about doing this?

(the file already exists right? are you meant to make a separate PR that only adds this line manually, after merging the other one?)

**Review by freben — APPROVED** [review]
Nice! Just some minor text stuff

**Review by benjdlambert — APPROVED** [review]
